ee/experimental/incident_utils.py

```diff
@@ -0,0 +1,148 @@
+import numpy as np
+import pandas as pd
+import networkx as nx
+
+from typing import List
+
+from keep.api.models.db.alert import Alert
+
+
+def mine_incidents(alerts: List[Alert], incident_sliding_window_size: int=6*24*60*60, statistic_sliding_window_size: int=60*60, 
+                   jaccard_threshold: float=0.0, fingerprint_threshold: int=1):
+    """
+        Mine incidents from alerts.
+    """
+    
+    alert_dict = {
+        'fingerprint': [alert.fingerprint for alert in alerts],
+        'timestamp': [alert.timestamp for alert in alerts],
+    }
+    alert_df = pd.DataFrame(alert_dict)
+    mined_incidents = shape_incidents(alert_df, 'fingerprint', incident_sliding_window_size, statistic_sliding_window_size, 
+                                      jaccard_threshold, fingerprint_threshold)
+    
+    return [
+        {
+            "incident_fingerprint": incident['incident_fingerprint'],
+            "alerts": [alert for alert in alerts if alert.fingerprint in incident['alert_fingerprints']],
+        }
+        for incident in mined_incidents
+    ]
+
+
+def get_batched_alert_counts(alerts: pd.DataFrame, unique_alert_identifier: str, sliding_window_size: int) -> np.ndarray:
```

Comment:
why does mine_incidents gets alerts a list[AlertDto] and here alerts are pd.DataFrame?